software/script/chameleon_cmd.py

```diff
@@ -810,44 +982,53 @@ def set_ble_connect_key(self, key: str):
         if len(data_bytes) != 6:
             raise ValueError("The ble connect key length must be 6")
 
-        return self.device.send_cmd_sync(DATA_CMD_SET_BLE_CONNECT_KEY_CONFIG, 0x00, data_bytes)
+        data = struct.pack('6s', data_bytes)
+        return self.device.send_cmd_sync(DATA_CMD_SET_BLE_PAIRING_KEY, data)
 
-    def get_ble_connect_key(self):
+    @expect_response(chameleon_status.Device.STATUS_DEVICE_SUCCESS)
+    def get_ble_pairing_key(self):
         """
         Get config of ble connect key
         """
-        return self.device.send_cmd_sync(DATA_CMD_GET_BLE_CONNECT_KEY_CONFIG, 0x00, None)
+        return self.device.send_cmd_sync(DATA_CMD_GET_BLE_PAIRING_KEY)
 
+    @expect_response(chameleon_status.Device.STATUS_DEVICE_SUCCESS)
     def delete_ble_all_bonds(self):
         """
         From peer manager delete all bonds.
         """
-        return self.device.send_cmd_sync(DATA_CMD_DELETE_ALL_BLE_BONDS, 0x00, None)
+        return self.device.send_cmd_sync(DATA_CMD_DELETE_ALL_BLE_BONDS)
 
+    @expect_response(chameleon_status.Device.STATUS_DEVICE_SUCCESS)
     def get_device_capabilities(self):
         """
-        Get (and set) commands that client understands
+        Get list of commands that client understands
         """
-
-        commands = []
-
         try:
-            ret = self.device.send_cmd_sync(DATA_CMD_GET_DEVICE_CAPABILITIES, 0x00)
-            self.device.commands = struct.unpack(f"{len(ret.data) // 2}H", ret.data)
-        except:
-            print("Chameleon doesn't understand get capabilities command. Please update firmware")
-
-        return commands
+            resp = self.device.send_cmd_sync(DATA_CMD_GET_DEVICE_CAPABILITIES)
+        except chameleon_com.CMDInvalidException:
+            print("Chameleon does not understand get_device_capabilities command. Please update firmware")
+            return chameleon_com.Response(cmd=DATA_CMD_GET_DEVICE_CAPABILITIES,
+                                          status=chameleon_status.Device.STATUS_NOT_IMPLEMENTED)
+        else:
+            if resp.status == chameleon_status.Device.STATUS_DEVICE_SUCCESS:
+                resp.data = [x[0] for x in struct.iter_unpack('!H', resp.data)]
+            return resp
 
-    def get_device_type(self):
+    @expect_response(chameleon_status.Device.STATUS_DEVICE_SUCCESS)
+    def get_device_model(self):
         """
-        Get device type
+        Get device model
         0 - Chameleon Ultra
```

Comment:
typo here, `1 - Chameleon Ultra`, `0 - Chameleon Lite`

Reply:
Should be correct with 0: ultra and 1: lite. 
See https://github.com/RfidResearchGroup/ChameleonUltra/blob/98605be5b8cf93bf6e54e8f84dab960dd8879f9a/firmware/common/hw_connect.h#L12